crates/telio-firewall/src/firewall.rs

```diff
@@ -2886,4 +3036,62 @@ pub mod tests {
         assert_eq!(tcppkgs[2].get_destination(), 12345);
         assert_eq!(tcppkgs[2].get_sequence(), 1);
     }
+
+    #[test]
+    fn firewall_udp_conns_reset() {
+        let fw = StatefullFirewall::new(false, false);
+        let peer = make_peer();
+        fw.add_to_port_whitelist(PublicKey(peer), FILE_SEND_PORT);
+
+        let test_udppkg = make_udp("127.0.0.2:12345", "101.101.101.101:54321");
+        // Outbound connection
+        assert!(fw.process_outbound_packet(&peer, &test_udppkg,));
+
+        #[derive(Default)]
+        struct Sink4 {
+            pkgs: Vec<Vec<u8>>,
+        }
+
+        impl io::Write for Sink4 {
+            fn write(&mut self, buf: &[u8]) -> io::Result<usize> {
+                self.pkgs.push(buf.to_vec());
+                Ok(buf.len())
+            }
+
+            fn flush(&mut self) -> io::Result<()> {
+                Ok(())
+            }
+        }
+
+        let mut sink4 = Sink4::default();
+        fw.reset_connections(&PublicKey(peer), None, &mut sink4, &mut std::io::sink()); // We do not test IPv6 yet
+
+        assert_eq!(sink4.pkgs.len(), 1); //3 packets in random order
```

Comment:
the comment says 3 packets but the code asserts for 1. Comment looks stale :)

Reply:
It is. Thanks, I will change the comment

Reply:
Done